src/Aspire.Hosting/Utils/ContainerReferenceParser.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Text.RegularExpressions;
+
+namespace Aspire.Hosting.Utils;
+
+/// <summary>
+/// Class to parse container references (e.g. "mcr.microsoft.com/dotnet/sdk:8.0")
+/// </summary>
+internal partial class ContainerReferenceParser
+{
+    public static ContainerReference Parse(string input)
+    {
+        if (string.IsNullOrEmpty(input))
+        {
+            throw new ArgumentOutOfRangeException(nameof(input), "repository name must have at least one component");
+        }
+
+        var match = ImageNameRegex().Match(input);
+
+        if (!match.Success)
+        {
+            throw new ArgumentOutOfRangeException(nameof(input), input, "invalid reference format");
```

Comment:
The error message isn't great, but it matches the error docker gives off, so 
https://github.com/distribution/reference/blob/8c942b0459dfdcc5b6685581dd0a5a470f615bff/reference.go#L72

Reply:
Maybe something like "invalid reference format: could not parse container image name"?